### PR TITLE
(done) Migrate some terrains that should be furnitures into furniture

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary

SUMMARY: Infrastructure "Changes some awkward terrains into furnitures for better layering"

#### Purpose of change
A great many things that are currently terrrains should in fact be furniture. For anything sticking above the ground, furniture has a lot more options for layering, preserving the identity of the terrain beneath it.

#### Describe the solution
This converts several manufactured/industrial terrains to furniture, obsoletes the old entries, and updates any palettes in the repository.

New furnitures;
- f_gas_tank
- f_diesel_tank
- f_gas_tank_smashed
- f_diesel_tank_smashed
- f_generator_broken
- f_machinery_light
- f_machinery_heavy
- f_machinery_electronic
- f_machinery_old
- f_console
- f_console_broken
- f_centrifuge
- f_compact_ASRG_containment (from t_plut_generator)

The compact ASRG containment unit has had the most changes from the old plut_generator. It now has a heavy lead and ceramic housing that can be removed to reveal a fragile kilopower-style nuclear reactor inside. It cannot be harvested for plutonium at this time but it can be smashed if you like being blown up and irradiated. Note that the explosion is a sodium reaction, not a nuclear one, you could survive it.

I have also adjusted some deconstruction recipes to include new ingredients available, and made the a "connects_to": "COUNTER" group for things like the consoles. When dismantled, consoles will leave behind a counter.

I have done a bit of palette updating while in maps, mostly to update region groundcover, and in a few places adding a fill_ter or other minor infrastructure fixes.  It bloated an already big PR but if I'm going to read several hundred palettes, I may as well fix stuff while I'm at it: these are things that would probably never get done otherwise.

#### Describe alternatives you've considered
Eventually we'd like to have more mapping layers than just terrain and furniture, but this is an early step towards that anyway.

#### Testing
image

Game runs without errors. Maps appear to spawn with the new furnitures instead of the old terrains. Consoles appear to work as intended.
I have tried to find some hard coded locations to see how it's working there, I couldn't find any I am sure of.
Can't think of anything else to test. Tired of looking at this PR.
image

### CODE REVIEW

File: data/json/furniture_and_terrain/furniture-storage.json [hunk 1]
```diff
@@ -878,6 +878,100 @@
       "items": [ { "item": "scrap", "count": [ 8, 32 ] }, { "item": "water_faucet", "prob": 50 } ]
     }
   },
+  {
+    "type": "terrain",
```
Comment (nexusmrsep) [inline data/json/furniture_and_terrain/furniture-storage.json:882]: I'm pretty certain you wanted `"type": "furniture"`  here, and in following additions.

**Comment by I-am-Erk** [comment]
Oh thank goodness. Thanks Kevin

**Comment by ZhilkinSerg** [comment]
Probably need tileset jsons updates - https://github.com/CleverRaven/Cataclysm-DDA/issues/42125

**Comment by I-am-Erk** [comment]
People should just use better, more up to date tilesets.

**Comment by Qrox** [comment]
Consoles in gas stations from old saves aren't working anymore (`iexamine::getNearFilledGasTank` checks for nearby gas tanks using furniture ids, but gas tanks from legacy saves aren't migrated to furnitures)

File: src/submap.cpp [hunk 2]
```diff
@@ -192,7 +193,7 @@ computer *submap::get_computer( const point &p )
 {
     // need to update to std::map first so modifications to the returned object
     // only affects the exact point p
-    update_legacy_computer();
+    //update_legacy_computer();
```
Comment (Qrox) [inline src/submap.cpp:196]: Why is this commented out? This might cause problems for older saves.

File: src/submap.cpp [hunk 3]
```diff
@@ -160,7 +160,7 @@ void submap::update_legacy_computer()
     if( legacy_computer ) {
         for( int x = 0; x < SEEX; ++x ) {
             for( int y = 0; y < SEEY; ++y ) {
-                if( ter[x][y] == t_console ) {
+                if( frn[x][y] == furn_str_id( "f_console" ) ) {
```
Comment (Qrox) [inline src/submap.cpp:163]: If it's a legacy computer, then the save may still contain "t_console" which should also be updated, unless we actually migrate "t_console" to "f_console" while loading the save.

**Pull request merged**